File: EnglishLearning.Client/src/types/notification.ts
export interface Notification {
  id: string
  userId: string
  type: NotificationType
  title: string
  message: string
  isRead: boolean
  data?: string
  createdAt: string
}

export type NotificationType =
  | 'QuizAssigned'
  | 'QuizStartingSoon'
  | 'QuizStarted'
  | 'QuizEnded'
  | 'QuizResultAvailable'

File: EnglishLearning.Domain/Entities/Notification.cs
using EnglishLearning.Domain.Common;
using EnglishLearning.Domain.Enums;

namespace EnglishLearning.Domain.Entities;

public class Notification : BaseEntity
{
    public string UserId { get; set; } = string.Empty;
    public NotificationType Type { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public bool IsRead { get; set; } = false;
    public string? Data { get; set; }

    public Notification() { }
}
